packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -24,16 +25,59 @@ import 'code_signing.dart';
 import 'ios_workflow.dart';
 import 'mac.dart';
 
+/// [IOSDeploy] is responsible for spawning and managing `ios-deploy` processes.
+///
+/// When an app is launched via `ios-deploy`, we wait for the `ios-deploy` process
+/// until the app has been started via LLDB. Once it has started, `ios-deploy` puts LLDB
+/// into "autopilot" mode, and we continue.
+///
+/// If this (Flutter tools) process terminates before `ios-deploy` or the app, `ios-deploy`
+/// will be suspended. This is problematic because it will keep the app process on the phone
+/// alive via LLDB. Therefore we kill any previously created `ios-deploy` process which
+/// has launched an app before trying to run it again.
 class IOSDeploy {
   const IOSDeploy();
 
+  io.File _tempFile(ApplicationPackage package) => io.File('/tmp/${package.id}.ios_deploy.pid');
```

Comment:
Can this use the injected filesystem `fs` instead of `dart:io` directly?

Reply:
Also, importing `dart:io` is causing CI to fail

Reply:
Sorry, I wasn't familiar with how the dependency injection worked.
I've fixed it.